src/LcdMenu.h

```diff
@@ -1,218 +1,244 @@
+     */
+    uint8_t maxCols;
+    /**
+     * Array of menu items
+     */
+    MenuItem* currentMenuTable;
+    /**
+     * Down arrow (↓)
+     */
+    byte downArrow[8] = {
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b10101,  // * * *
+        0b01110,  //  ***
+        0b00100   //   *
+    };
+    /**
+     * Up arrow (↑)
+     */
+    byte upArrow[8] = {
+        0b00100,  //   *
+        0b01110,  //  ***
+        0b10101,  // * * *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100,  //   *
+        0b00100   //   *
+    };
+    /**
+     * Draws the cursor
+     */
+    void drawCursor();
+    /**
+     * Draw the menu items with up and down indicators
+     */
+    void drawMenu();
+    /**
+     * Check if the cursor is at the start of the menu items
+     *
+     * @return true : `boolean` if it is at the start
+     */
+    boolean isAtTheStart();
+    /**
+     * Check if the cursor is at the end of the menu items
+     *
+     * @return true : `boolean` if it is at the end
+     */
+    boolean isAtTheEnd();
+    /*
+     * Draw the menu items and cursor
+     */
+    void paint();
+    /**
+     * Reset the display
+     */
+    void reset();
+    /**
+     * Places the cursor at end of Menu's text.
+     *
+     * @param item MenuItem where the cursor should be placed
+     * @relatesalso MenuItem
+     */
+    void placeCursorAtEnd(MenuItem* item);
+
+   public:
+    /**
+     * Time when the toast started showing in milliseconds
+     */
+    unsigned long startTime;
+    /**
+     * How long the toast should Last in milliseconds
+     */
+    unsigned int duration;
+    /**
+     * LCD Display
+     */
+    
+#ifdef LCD_NO_I2C
+    LiquidCrystal* lcd;
+#else
+    LiquidCrystal_I2C* lcd;
+#endif
+    /**
+     * Constructor for the LcdMenu class
+     *
+     * @param maxRows rows on lcd display e.g. 4
+     * @param maxCols columns on lcd display e.g. 20
+     * @return new `LcdMenu` object
+     */
+    LcdMenu(uint8_t maxRows, uint8_t maxCols);
+    /**
+     * Call this function in `setup()` to initialize the LCD and the custom
+     * characters used as up and down arrows
+     *
+     * @param lcd_Addr address of the LCD on the I2C bus (default 0x27)
+     * @param menu menu to display
+     */
+    void setupLcdWithMenu(     
+#ifdef LCD_NO_I2C
+       uint8_t rs, uint8_t en, uint8_t d0, uint8_t d1, uint8_t d2, uint8_t d3, MenuItem *menu) {
+      lcd =  new LiquidCrystal(rs, en, d0, d1, d2, d3);
+      lcd->begin(maxCols, maxRows);
+#else
+       uint8_t lcd_Addr, MenuItem *menu) {
+      lcd =  new LiquidCrystal_I2C(lcd_Addr, maxCols, maxRows);
+      lcd->init();
+      lcd->backlight();
+#endif
+      lcd->clear();
+      lcd->createChar(0, upArrow);
+      lcd->createChar(1, downArrow);
+      this->currentMenuTable = menu;
+      paint();
+   }

```

Comment:
Could you instead place this function's definition in the `LcMenu.cpp` file, the header should contain only function declaration. You can see that the other functions have just been declared.

Reply:
I can only merge up to this commit, [REASON](https://github.com/forntoh/LcdMenu/pull/3#pullrequestreview-463263743)
The other commits violate the original intentions of the library.

So rather than deleting stuff, you can add an option to use the keypad shield, but must always give the freedom to the user.